src/main/java/com/greghaskins/spectrum/internal/CompositeTest.java

```diff
@@ -26,18 +26,15 @@ public boolean isAtomic() {
     return true;
   }
 
-  private static void abortOnFailureChildRunner(final Suite suite, final RunNotifier runNotifier) {
-    FailureDetectingRunListener listener = new FailureDetectingRunListener();
-    runNotifier.addListener(listener);
-    try {
-      for (Child child : suite.children) {
-        if (listener.hasFailedYet()) {
-          child.ignore();
-        }
-        suite.runChild(child, runNotifier);
+  private static void abortOnFailureChildRunner(final Suite suite,
+      final RunReporting<Description, Failure> reporting) {
+    FailureDetectingRunDecorator<Description, Failure> decoratedReporting =
```

Comment:
This decorator simplifies the method a lot. We no longer add a listener to the `RunNotifier` but instead wrap the `Reporting` with a sentinel.